### DISCUSSION TITLE
Random dot kinematogram leaves visible traces

### DISCUSSION THREAD

**Discussion by ThetaS** [bonsai-rx/bonsai#1001] [Q&A]
Hi all,

When implementing dot kinematogram with C Sharp Transform in Bonsai, I found that the dots will leave a visible trace as they move. I checked the RDK implemented in the Bonsai official scripting tutorial and it also has this problem. I suppose it's due to some buffering or frame refresh rate issues. Any ideas on how to solve this problem? Thanks.

**Comment by bruno-f-cruz** [comment]
Can you include your project? Thanks!

**Comment by ThetaS** [comment]
Hi Bruno,

Sorry for suspending this for reallly long time. Attached is my Bonsai layout and C# code. I also have question regarding how to construct this kind figure-ground visual stimulus in general, especially using the default "Draw Grating" scheme. It seems like the draw grating does not have a mask option. Do I need to code it from scratch suppose I want to have two gratings  overlay on each other with one as foreground and one as background? Thanks a lot.
[RDK.zip](https://github.com/bonsai-rx/bonsai/files/9539334/RDK.zip)

**Reply by bruno-f-cruz** [reply]
I think using `RenderFrame` instead of `UpdateFrame` to generate your RDK might fix it.

**Reply by ThetaS** [reply]
Thanks, will try to implement that
